### PR TITLE
wire element was modified

### PR DESCRIPTION
**Pull request by anpatape**

### CODE REVIEW

File: SixTrack/sixtrack.s [hunk 1]
```diff
@@ -23552,143 +23357,84 @@ C     Convert r(1), r(2) from U(0,1) -> rvec0 as Gaussian with cutoff mcut (#sig
...
 *FOX  (ONE+DPDA)-YY(1)*YY(1)-YY(2)*YY(2)))-TX)) ;
 *FOX  YY(1)=SQRT((ONE+DPDA)*(ONE+DPDA)-YY(2)*YY(2))*SIN(ATAN(YY(1)/
 *FOX  SQRT((ONE+DPDA)*(ONE+DPDA)-YY(1)*YY(1)-YY(2)*YY(2)))-TX) ;
-*FOX  XX(1)=XX(1)-XX(2)*SIN(TY)*YY(1)/SQRT((ONE+DPDA)*(ONE+DPDA)-
+
+*FOX  XI=XI-YI*SIN(TY)*YY(1)/SQRT((ONE+DPDA)*(ONE+DPDA)-
 *FOX  YY(1)*YY(1))/COS(ATAN(YY(2)/SQRT((ONE+DPDA)*(ONE+DPDA)-
 *FOX  YY(1)*YY(1)-YY(2)*YY(2)))-TY) ;
-*FOX  XX(2)=XX(2)*(COS(TY)-SIN(TY)*TAN(ATAN(YY(2)/SQRT((ONE+DPDA)*
+*FOX  YI=YI*(COS(TY)-SIN(TY)*TAN(ATAN(YY(2)/SQRT((ONE+DPDA)*
```
Comment (anpatape) [inline SixTrack/sixtrack.s]: why is more green?

File: Doc/physics_manual/sixphys.tex [hunk 2]
```diff
@@ -802,45 +802,93 @@ \subsection{Phase-trombone, matrix element}
...
+ \begin{align*}
+ XI &  \to x-dx &
+ YI &  \to y-dy
+ \end{align*}
+
+ Step 2. Symplectic rotation (\cite{forest99}) 
  
+ Rotation on $\theta$ (OX tilt):
  \begin{align}
+p_z & \to \sqrt{1-p_x^2-p_y^2} \\
```
Comment (rdemaria) [inline Doc/physics_manual/sixphys.tex]: The equation does not look correct pz=sqrt((1+delta)^2 -px^2 - py^2))

File: Doc/physics_manual/sixphys.tex [hunk 3]
```diff
@@ -802,45 +802,93 @@ \subsection{Phase-trombone, matrix element}
...
- 
- \begin{align}
-     x' &\to x' - \frac{\frac{cur\cdot10^{-7}}{chi}\cdot x_i}{x_i^2+y_i^2}
-     \left[\sqrt{(3/2 l)^2+x_i^2+y_i^2}-\sqrt{(l/2)^2+x_i^2+y_i^2} \right] \\
-     y' &\to y' - \frac{\frac{cur\cdot10^{-7}}{chi}\cdot y_i}{x_i^2+y_i^2}
-     \left[\sqrt{(3/2)^2+x_i^2+y_i^2}-\sqrt{(l/2)^2+x_i^2+y_i^2} \right]
+
+Rotation on $\phi$ (OY tilt):
+\begin{align}
+p_z & \to \sqrt{1-p_x^2-p_y^2} \\
```
Comment (rdemaria) [inline Doc/physics_manual/sixphys.tex]: pz not correct,  see  above

File: SixTrack/sixtrack.s [hunk 4]
```diff
@@ -8421,374 +8422,165 @@ cc2008
...
+!     e -> 1; m0/4Pi -> 1.0e-7; N -> 1.0e-7*I
 !     magnetic rigidity
-!hr03 chi = sqrt(e0*e0-pmap*pmap)*c1e6/clight
-      chi = (sqrt(e0**2-pmap**2)*c1e6)/clight                            !hr03
 
-      ix = ixcav
+      chi = (sqrt(e0**2-pmap**2)*c1e6)/clight                            
+
+!     
       tx = xrms(ix)
```
Comment (rdemaria) [inline SixTrack/sixtrack.s]: what is xrms, zrms? why you change tx and then assign it to Tx?

**Comment by kyrsjo** [comment]
TODO:
 * Finish?
 * Test
 * Merge current upstream/master into this branch.
 * Review...

**Comment by kyrsjo** [comment]
Once you've done your tests and merged master onto your branch, let's discuss how to finish this merge.

Important thing:
SixTest (Miriam also knows how): Please make a new sub-folder `wire` in SixTest with input files fort.2, fort.3, fort.8 and fort.16 (just those you need)) for running SixTrack with the wire element. Commit these files to GIT. If you need more files, put and commit them also, and make a file extra_inputs.txt containing the names of the extra input files (one file per line).

Then set enviroment variable EXECS=/path/to/SixTrack/binary, TESTS=wire, and run ./run pro with some numerical argument. Once it is finished, you will see that you have files "fort.10.SixTrack_blahblahba_yournumber_blablabla", and similarly fort.90 and fort.6. Rename these to fort.10.canonical, fort.90.canonical and fort.6.canonical and commit (assuming they are not 10s of MB, if so we need to discuss). You can now test with ./check_10 yournumber and ./check_90 yournumber. If you have more output files, make a file "extra_checks.txt" containing the filenames and the command to check -- see the `fma` check for an example.

Another important thing:
Do you know how to deal with CRLIBM & numerical reproducibility, and checkpoint/restart (including lout/normal write)? Do you have any new variables which change during the run (and thus need to be checkpointed)?

**Comment by mfittere** [comment]
We will do all that one by one! Thanks for the check list :).


Sent from my mobile phone.



On Fri, Dec 16, 2016 at 10:53 AM -0600, "Kyrre Ness Sjøbæk" <notifications@github.com<mailto:notifications@github.com>> wrote:


Once you've done your tests and merged master onto your branch, let's discuss how to finish this merge.

Important thing:
SixTest (Miriam also knows how): Please make a new sub-folder wire in SixTest with input files fort.2, fort.3, fort.8 and fort.16 (just those you need)) for running SixTrack with the wire element. Commit these files to GIT. If you need more files, put and commit them also, and make a file extra_inputs.txt containing the names of the extra input files (one file per line).

Then set enviroment variable EXECS=/path/to/SixTrack/binary, TESTS=wire, and run ./run pro with some numerical argument. Once it is finished, you will see that you have files "fort.10.SixTrack_blahblahba_yournumber_blablabla", and similarly fort.90 and fort.6. Rename these to fort.10.canonical, fort.90.canonical and fort.6.canonical and commit (assuming they are not 10s of MB, if so we need to discuss). You can now test with ./check_10 yournumber and ./check_90 yournumber. If you have more output files, make a file "extra_checks.txt" containing the filenames and the command to check -- see the fma check for an example.

Another important thing:
Do you know how to deal with CRLIBM & numerical reproducibility, and checkpoint/restart (including lout/normal write)? Do you have any new variables which change during the run (and thus need to be checkpointed)?

**Comment by mfittere** [comment]
Questions:
- The wire is also considered in the differential algebra part (look for wireda). Is there a test for this part???
- there are two options to define the distance of the wire:
  - in respect to the closed orbit (-15)
  - in respect to the reference frame (15)
  Do we want to keep this definition or should we introduce a WIRE block in fort.3 in order to be able to define the variables in a more understandable way as done for example for the elens in the ELEN block?

**Comment by rdemaria** [comment]
Can you make a proposal for the block? For the test it is enough that the tune 6d is changed by the wire in the correct way. Will you investigate the issue with the closed orbit?

**Comment by mfittere** [comment]
There is no issue with the closed orbit any more - Andrei fixed it :-)!

What do you mean with the tune6D?

**Comment by rdemaria** [comment]
Simply that the tune calculated in 6D (standard procedure) use the DA code.

**Comment by mfittere** [comment]
Where is this calculated? Do you know the part in the code?

**Comment by mfittere** [comment]
Proposal for wire block:
```
WIRE
wire1 current length integrated_length closed_orbit_flag displacement_x displacement_y tilt_x tilt_y
```

There is another option for the BEAM block. The wire consistently treats the `ibeco` parameter as done for bb. We thought that this is probably saver than introducing another flag in the WIRE block as you either want to simulate the bb+wire with the closed orbit kick or not, so both should be treated in the same way.

**Comment by rdemaria** [comment]
Why do you have the close orbit flag if the wire is honouring ibeco?

**Comment by mfittere** [comment]
I understood that the:
- ibeco: controls if the closed orbit kick from the wire is taken into account
- closed_orbit_flag: controls if the distance (displacement_*) is taken in respect to the reference frame (x=y=0) or the closed orbit

**Comment by rdemaria** [comment]
Ok, clear!

On 21 Dec 2016 20:11, "mfittere" <notifications@github.com> wrote:

I understood that the:

   - ibeco: controls if the closed orbit kick from the wire is taken into
   account
   - closed_orbit_flag: controls if the distance (displacement_*) is taken
   in respect to the reference frame (x=y=0) or the closed orbit

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/SixTrack/SixTrack/pull/20#issuecomment-268612938>, or mute
the thread

**Comment by kyrsjo** [comment]
Hi, I'm working on updating this branch with the current master, and various cleanups; see https://github.com/SixTrack/SixTrack/compare/master...kyrsjo:wire_merge . I'l make a pull request to the anpatape:wire branch (i.e. into this PR) once I'm done.

One thing that is definitively missing is the user manual!

**Comment by mfittere** [comment]
Kyrre, wait!!!!

We are working on this and have a plan. Can you give us some time.

Miriam

**Comment by kyrsjo** [comment]
Thanks for the tests!

But could you please only upload relevant test files, i.e. inputs fort.2, fort.3, fort.8, and fort.16, and outputs fort.10(.canonical) and fort.90(.canonical)? This is what is used by the test program run_pro. If you want more inputs and outputs than that, you'll need the newer version of SixTest we wrote for FMA etc. + extra_inputs and extra_tests files...

**Comment by mfittere** [comment]
Hi Kyrre,

yes, we are just doing that. We managed to update the changes in 
sixtrack.s and merge with your branch. Andrei is just updating the 
tests, then we will clean up the files and basically overwrite the tests 
(folders wire* only) in your wire_merge branch. After that I try to 
merge my changes and then we should have hopefully the first running 
version.

Don't worry, everything (still ;-) ) under control here.

Miri

**Comment by mfittere** [comment]
Kyrre? What about this new hugenpart file in the test folders?

**Comment by kyrsjo** [comment]
For some tests, I also compiled with the flag stf and generated an STF canonical. This is an alternative to fort.90. I can add this before merging...

**Comment by kyrsjo** [comment]
Regarding the wire, it would be nice if the commons follow some naming convention, as is done for e-lens, DYNK, etc. ...

**Comment by mfittere** [comment]
I have that in my branch. They are called wire_*. In the tracking 
routine they are still messy. Once I have merged my changes to the new 
wire, I can work on that, if there is still time. But it is on the list ;-).

**Comment by kyrsjo** [comment]
Also dead branch, discussion now in #174

**Pull request closed without merging**